src/read/macros.rs

```diff
@@ -97,23 +98,267 @@ impl<R> From<R> for DebugMacInfo<R> {
+        DebugMacro {
+            section: macro_section,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct MacroUnitHeader<R: Reader> {
+    /// The version of the macro unit header. At the moment only version 5 is defined.
+    _version: u16,
+    flags: u8,
+    debug_line_offset: DebugStrOffsetsBase<R::Offset>,
+}
+
+impl<R: Reader> MacroUnitHeader<R> {
+    const OFFSET_SIZE_FLAG: u8 = 0b0000_0001;
+    const DEBUG_LINE_OFFSET_FLAG: u8 = 0b0000_0010;
+    const OPCODE_OPERANDS_TABLE_FLAG: u8 = 0b0000_0100;
+
+    fn parse(input: &mut R) -> Result<Self> {
+        let version = input.read_u16()?;
+        let flags = input.read_u8()?;
+        let format = if flags & Self::OFFSET_SIZE_FLAG == 0 {
+            Format::Dwarf32
+        } else {
+            Format::Dwarf64
+        };
+        let debug_line_offset = if flags & Self::DEBUG_LINE_OFFSET_FLAG != 0 {
+            DebugStrOffsetsBase(input.read_offset(format)?)
+        } else {
+            DebugStrOffsetsBase(R::Offset::from_u64(0)?)
+        };
+        // if the opcode operands table flag is set, there is a table in the header which currently isn't parsed
+        if flags & Self::OPCODE_OPERANDS_TABLE_FLAG != 0 {
+            return Err(Error::UnsupportedOpcodeOperandsTable);
+        }
+        Ok(MacroUnitHeader {
+            _version: version,
+            flags,
+            debug_line_offset,
+        })
+    }
+
+    fn format(&self) -> Format {
+        if self.flags & Self::OFFSET_SIZE_FLAG == 0 {
+            Format::Dwarf32
+        } else {
+            Format::Dwarf64
+        }
+    }
+}
+
+/// A string in a macro entry.
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub enum DebugMacInfoItem<R> {
+pub enum MacroString<R, Offset = <R as Reader>::Offset>
+where
+    R: Reader<Offset = Offset>,
+    Offset: ReaderOffset,
+{
+    /// The string is directly embedded in the macro entry
+    Direct(R),
+    /// The macro refers to a string in the `.debug_str` section using a `DebugStrOffset`.
+    StringPointer(DebugStrOffset<Offset>),
+    /// The macro contains an index into an array in the `.debug_str_offsets`
+    /// section, which refers to a string in the `.debug_str` section.
+    IndirectStringPointer(DebugStrOffsetsBase<Offset>, DebugStrOffsetsIndex<Offset>),
+    /// The macro refers to a string in the `.debug_str` section in the supplementary object file
+    Supplementary(DebugStrOffset<Offset>),
+}
+
+impl<R: Reader> MacroString<R> {
+    /// Get the string slice from the macro entry.
+    pub fn string(&self, unit: UnitRef<'_, R>) -> Result<R> {
+        match self {
+            MacroString::Direct(s) => Ok(s.clone()),
+            MacroString::StringPointer(offset) => unit.string(*offset),
+            MacroString::IndirectStringPointer(base, index) => {
+                let str_offset = unit.dwarf.debug_str_offsets.get_str_offset(
+                    unit.header.format(),
+                    *base,
+                    *index,
+                )?;
```

Comment:
```suggestion
                let str_offset = unit.str_offset(*index)?;
```
It's a bit unclear in the standard, but my expectation is that it should be this.